web/src/app/ui/kmwuitoggle.ts

```diff
@@ -23,682 +34,756 @@ if(!window['keyman']['ui']['name']) {
    **/
 
   try {
-
     // Declare KeymanWeb, OnScreen Keyboard and Util objects
-    var keymanweb=window['keyman'],util=keymanweb['util'];
-    var dbg=keymanweb['debug'];
+    const keymanweb = keyman;
+    //var dbg=keymanweb['debug'];
+    const util = keymanweb.util;
 
     // Disable UI for touch devices
-    if(util['isTouchDevice']()) throw '';
-
-    // Initialize user interface common variables
-    var ui:any=keymanweb['ui'] = {
-      name: 'toggle',
-      initialized: false,
-      controller: null,
-      oskButton: null,
-      kbdButton: null,
-      controllerHovered: false,
-      keyboards: [],
-      lastActiveKeyboard: -1,
-      selectedMenuItem: null,
-      updateList: true,
-      updateTimer: null
+    if(util.isTouchDevice()) {
+      throw '';
     }
 
-    /**
-     * Update the KeymanWeb user interface when an input element is focused or blurred
-     *
-     * @param       {Object}            someElement     focused element
-     * @param       {(boolean|number)}  focusing        true if focusing
-     * @param       {Object}            activeControl   Object representing API specs for the control, if it exists and is now focused.
-     */
-    ui.doFocus = function(someElement,focusing, activeControl) {
-      // This callback must be ignored until UI is initialized, or for touch devices (which can never initialize the UI)
-      if(!ui.initialized) {
-        return;
-      }
-
-      // We don't want to shift the controller to something that's not an input element,
-      // but do want to account for window.event's data when legitimate.
-      if(window.event && keymanweb['isAttached'](window.event.srcElement)) {
-        someElement=window.event.srcElement;
-      }
+    class ToggleUI implements UIModule {
+      // Initialize user interface common variables
+      readonly name = 'toggle';
+      initialized = false;
+
+      /**
+       * The top-level element of the Toggle UI module.
+       */
+      controller: HTMLDivElement = null;
+      oskButton: ReturnType<ToggleUI['button']> = null;  // inner class - Button, as defined within the .button method
+      kbdButton: ReturnType<ToggleUI['button']> = null;  // likewise.
+      controllerHovered = false;
+
+      keyboards: KeyboardMenuEntry[] = [];
+      /**
+       * The index of the previously-active keyboard in this UI module's language list.
+       */
+      lastActiveKeyboard = -1;
+
+      selectedMenuItem: HTMLAnchorElement = null;
+      updateList = true;
+
+      /**
+       * The numeric handle for an active window.setTimeout call.
+       */
+      updateTimer: number = null;
+
+      keyboardMenu: HTMLUListElement;
```

Comment:
This one was utilized inline, but was never declared with the other class fields in the original version.  Now that we can do proper type-checking here, the compiler wanted a proper declaration.